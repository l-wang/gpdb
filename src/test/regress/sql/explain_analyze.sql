-- GPDB_MERGE_12_FIXME: replace the match ignores with SUMMARY OFF
-- start_matchignore
-- m/Execution time: \d+\.\d+ ms/
-- m/Planning time: \d+\.\d+ ms/
-- end_matchignore

CREATE TEMP TABLE empty_table(a int);
-- We used to incorrectly report "never executed" for a node that returns 0 rows
-- from every segment. This was misleading because "never executed" should
-- indicate that the node was never executed by its parent.
-- explain_processing_off
EXPLAIN (ANALYZE, TIMING OFF, COSTS OFF) SELECT a FROM empty_table;
-- explain_processing_on
