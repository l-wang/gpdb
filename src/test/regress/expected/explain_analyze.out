-- GPDB_MERGE_12_FIXME: replace the match ignores with SUMMARY OFF
-- start_matchignore
-- m/Execution time: \d+\.\d+ ms/
-- m/Planning time: \d+\.\d+ ms/
-- end_matchignore
CREATE TEMP TABLE empty_table(a int);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'a' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
-- We used to incorrectly report "never executed" for a node that returns 0 rows
-- from every segment. This was misleading because "never executed" should
-- indicate that the node was never executed by its parent.
-- explain_processing_off
EXPLAIN (ANALYZE, TIMING OFF, COSTS OFF) SELECT a FROM empty_table;
                                   QUERY PLAN                                    
---------------------------------------------------------------------------------
 Gather Motion 3:1  (slice1; segments: 3) (actual rows=0 loops=1)
   Executor Memory: 0kB  Segments: 1  Max: 0kB (segment -1)
   ->  Seq Scan on empty_table (actual rows=0 loops=1)
         Executor Memory: 0kB  Segments: 3  Max: 0kB (segment 0)
 Planning time: 21.532 ms
   (slice0)    Executor memory: 36K bytes.
   (slice1)    Executor memory: 35K bytes avg x 3 workers, 35K bytes max (seg0).
 Memory used:  128000kB
 Optimizer: Pivotal Optimizer (GPORCA)
 Execution time: 1.045 ms
(8 rows)

-- explain_processing_on
